src/main/java/loci/common/Location.java

```diff
@@ -118,6 +118,44 @@
   private URI uri;
   private File file;
 
+  class URLLocationProperties {
```

Comment:
Longer-term, this should likely be at least a static inner class if not a top-level class.